### PR TITLE
titanium-search-element

### PR DESCRIPTION
**Pull request by kaseyhinton**

### CODE REVIEW

File: src/titanium-search-input.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  render() {
+    return html
+    `<input-container>
+  <svg search viewBox="0 0 24 24">
+    <path
+      d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z">
+    </path>
+  </svg>
+  <input type="text" ?disabled="${this.disabled}" placeholder="${
+        this.placeholder}" autocomplete="off" .value="${this.value}" @keyup="${
```
Comment (antwonthegreat) [inline src/titanium-search-input.ts]: I think there need to be more listeners or different listeners bound here. You can enter text without firing key-up by pasting, tabbing away, etc. Is there a @change you can bind to?

File: src/titanium-search-input.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    this._input.focus();
+  }
+
+  static styles = css`
+  :host {
+    display: block;
+  }
+
+  input-container {
+    @apply --layout-self-end;
```
Comment (aarondrabeck) [inline src/titanium-search-input.ts]: Lets not use iron-layout in these root components,

File: src/titanium-search-input.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    `<input-container>
+  <svg search viewBox="0 0 24 24">
+    <path
+      d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z">
+    </path>
+  </svg>
+  <input type="text" ?disabled="${this.disabled}" placeholder="${
+        this.placeholder}" autocomplete="off" .value="${this.value}" @keyup="${
+        this._onValueChange}" on />
+  <titanium-svg-button
```
Comment (aarondrabeck) [inline src/titanium-search-input.ts]: Can we add a attribute to hide this button?

File: src/titanium-search-input.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
```
Comment (aarondrabeck) [inline src/titanium-search-input.ts]: Can we add a attribute to hide this?

File: src/titanium-search-input.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
+import '../lib/titanium-svg-button';
+import {css, customElement, html, LitElement, property} from 'lit-element';
+
+@customElement('titanium-search-input')
+export class TitaniumSearchInput extends LitElement {
+  @property() value: string = '';
+  @property() placeholder: string = '';
+  @property({type: Boolean, reflect: true}) disabled: boolean;
+
+  _input: HTMLElement|any;
```
Comment (aarondrabeck) [inline src/titanium-search-input.ts]: change to `_input: HTMLInputElement;`

File: src/titanium-search-input.ts (hunk 1 above)
Comment (kaseyhinton) [reply src/titanium-search-input.ts]: Yeah, that is a good point. I will add change listener

File: src/titanium-search-input.ts (hunk 2 above)
Comment (kaseyhinton) [reply src/titanium-search-input.ts]: I will address this.

**Pull request merged**